### PR TITLE
[FLINK-19295][yarn][tests] Exclude more meaningless akka shutdown errors

### PR DESCRIPTION
**Pull request by rmetzger**
## What is the purpose of the change

A test stumbled across some exceptions from Akka's shutdown. This PR is adding them to the whitelist of accepted exceptions.

### CODE REVIEW

File: flink-yarn-tests/src/test/java/org/apache/flink/yarn/YarnTestBase.java [hunk 1]
```diff
@@ -128,10 +128,11 @@
 		// workaround for annoying InterruptedException logging:
 		// https://issues.apache.org/jira/browse/YARN-1022
 		"java.lang.InterruptedException",
-		// very specific on purpose
+		// very specific on purpose; whitelist meaningless exceptions that occur during akka shutdown:
 		"Remote connection to [null] failed with java.net.ConnectException: Connection refused",
 		"Remote connection to [null] failed with java.nio.channels.NotYetConnectedException",
 		"java.io.IOException: Connection reset by peer",
+		"Caused by: [java.net.ConnectException: Connection refused:",
```
Comment (zentol) [inline flink-yarn-tests/src/test/java/org/apache/flink/yarn/YarnTestBase.java:135]: I'm wondering whether we should rework this to work on `Patterns` instead; then we could also include `akka.remote.ReliableDeliverySupervisor`, or in fact the entire message.

File: flink-yarn-tests/src/test/java/org/apache/flink/yarn/YarnTestBase.java [hunk 2]
```diff
@@ -123,26 +124,26 @@
 	};
 
 	/** These strings are white-listed, overriding the prohibited strings. */
-	protected static final String[] WHITELISTED_STRINGS = {
-		"akka.remote.RemoteTransportExceptionNoStackTrace",
+	protected static final Pattern[] WHITELISTED_STRINGS = {
+		Pattern.compile("akka\\.remote\\.RemoteTransportExceptionNoStackTrace"),
```
Comment (zentol) [inline flink-yarn-tests/src/test/java/org/apache/flink/yarn/YarnTestBase.java:128]: nit: In practice `\\.` is probably not necessary here; chances are slim that another message would manage to match the pattern. You could also pass the `LITERAL` flag for all but the akka message.

**Review by zentol — APPROVED** [review]
+1

**Comment by rmetzger** [comment]
Thanks a lot for your review. Merging.

**Pull request merged**